### PR TITLE
u2ps: init at 1.2

### PR DESCRIPTION
**Pull request by athas**
## Description of changes

u2ps is text to postscript converter similar to a2ps, with emphasis on Unicode support.

It is stable and has few dependencies and should not be difficult to maintain in Nixpkgs.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/text/u2ps/default.nix
Comment (ius) [inline pkgs/tools/text/u2ps/default.nix:22]: U2ps is **a** text to postscript converter similar to a2ps ...

File: pkgs/tools/text/u2ps/default.nix
Comment (ius) [inline pkgs/tools/text/u2ps/default.nix:12]: Use `fetchFromGitHub` instead

File: pkgs/tools/text/u2ps/default.nix
Comment (ius) [inline pkgs/tools/text/u2ps/default.nix:4]: You're not using `autoconf` anywhere

**Review by ius — CHANGES_REQUESTED** [review]

File: pkgs/tools/text/u2ps/default.nix
Comment (athas) [reply pkgs/tools/text/u2ps/default.nix]: Removed.

File: pkgs/tools/text/u2ps/default.nix
Comment (athas) [reply pkgs/tools/text/u2ps/default.nix]: Done.

File: pkgs/tools/text/u2ps/default.nix
Comment (athas) [reply pkgs/tools/text/u2ps/default.nix]: Done.

**Review by ius — APPROVED** [review]

File: pkgs/tools/text/u2ps/default.nix
Comment (AndersonTorres) [inline pkgs/tools/text/u2ps/default.nix:7]: Let's use `finalAttrs`:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/tools/text/u2ps/default.nix
Comment (AndersonTorres) [inline pkgs/tools/text/u2ps/default.nix:14]:
```suggestion
    repo = "u2ps";
    rev = finalAttrs.version;
```

File: pkgs/tools/text/u2ps/default.nix
Comment (AndersonTorres) [inline pkgs/tools/text/u2ps/default.nix:31]:
```suggestion
})
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/tools/text/u2ps/default.nix
Comment (AndersonTorres) [inline pkgs/tools/text/u2ps/default.nix:29]: Set `meta.mainProgram` to the name of the "most important" executable.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/tools/text/u2ps/default.nix
Comment (athas) [reply pkgs/tools/text/u2ps/default.nix:30]: Done.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Finally, squash all commits into only one.

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/tools/text/u2ps/default.nix
Comment (drupol) [inline pkgs/tools/text/u2ps/default.nix:20]: Could you please sort the `meta` attributes alphabetically?

**Review by drupol — CHANGES_REQUESTED** [review]
Hi !

Thanks for your contribution.

I've left some comments, let me know if this is clear enough.

**Review by drupol — APPROVED** [review]
All good now! Waiting for the CI to be done before merging !

**Pull request merged**
